app.go

```diff
@@ -57,9 +57,8 @@ func New(opts ...Option) *App {
 		lifecycle: lifecycle,
 		logger:    logger,
 	}
-	app.Provide(func() Lifecycle {
-		return lifecycle
-	})
+	app.Provide(func() Lifecycle { return lifecycle })
+	app.Provide(func() SemanticVersion { return Version })
```

Comment:
Don't provide in container, it's a constant. Other things should just use it directly.